staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -604,9 +624,22 @@ func (r *Request) WatchWithSpecificDecoders(wrapperDecoderFn func(io.ReadCloser)
 		}
 		return nil, fmt.Errorf("for request %s, got status: %v", url, resp.StatusCode)
 	}
-	wrapperDecoder := wrapperDecoderFn(resp.Body)
+
+	contentType := resp.Header.Get("Content-Type")
+	mediaType, params, err := mime.ParseMediaType(contentType)
+	if err != nil {
+		klog.V(4).Infof("Unexpected content type from the server: %q: %v", contentType, err)
+	}
+	decoder, streamingSerializer, framer, err := r.c.content.Negotiator.StreamDecoder(mediaType, params)
+	if err != nil {
+		return nil, err
+	}
+
+	frameReader := framer.NewFrameReader(resp.Body)
+	streamingDecoder := streaming.NewDecoder(frameReader, streamingSerializer)
```

Comment:
is it unreasonable of me to be confused that `StreamDecoder()` returns `decoder`, from which we construct `streamingDecoder`, and then we pass both into `NewDecoder(...)`? can we name these better?

Reply:
I mean, we can.  They're the same names we use today.

Reply:
would these names be clearer (and accurately match use)?
* streamingDecoder -> watchEventDecoder
* decoder -> objectDecoder